keyserver/storage/interface.go

```diff
@@ -53,7 +53,7 @@ type Database interface {
 
 	// DeviceKeysForUser returns the device keys for the device IDs given. If the length of deviceIDs is 0, all devices are selected.
 	// If there are some missing keys, they are omitted from the returned slice. There is no ordering on the returned slice.
-	DeviceKeysForUser(ctx context.Context, userID string, deviceIDs []string) ([]api.DeviceMessage, error)
+	DeviceKeysForUser(ctx context.Context, userID string, deviceIDs []string, includeEmpty bool) ([]api.DeviceMessage, error)
```

Comment:
Flags like this aren't clear from callsites which makes it harder to know which to use. `includeEmpty` is also somewhat vague. What is an empty device?

Reply:
Honestly, I was surprised at the behaviour. It seems we create device key entries that just contain an empty key JSON when the device gets created. Then for some reason we filter them out until they get populated later by a key upload by the client. My guess is that it was so that the two tables stay roughly in sync between the user API and the key server, but I don't know why we bother with that — your guess is as good as mine.

Reply:
iirc we need some way to track deleted devices I think, which might be what this is.